modules/ROOT/pages/index.adoc

```diff
@@ -1,39 +1,217 @@
+
+You can design Mule apps in Anypoint Studio, in Design Center, or even with
+handwritten XML. Studio and Design Center UIs represent message processors as
+icons within flow components. Studio also represents them as XML elements in a
+Configuration XML.
+
+////
+The example below is a simple flow created in Anypoint Studio's visual editor;
+click the *XML Editor* tab to view the XML of a Mule project that holds the same
+flow.
+
+[.ex]
+=====
+[discrete.view]
+=== Studio Visual Editor
+
+image::simple-flow2.png[simple+flow2]
+
+[discrete.view]
+=== XML Editor
+
+[source,xml,linenums]
+----
+include::{examplesdir}/mule-concepts.xml[]
+----
+=====
+////
+
+For more on this topic, see
+xref:design-center/about-designing-a-mule-application.adoc[Mule Flow Design (Design Center)]
+and xref:7.2@studio:index.adoc[Anypoint Studio].
+
+[[dataweave]]
+== DataWeave
+
+DataWeave is the primary language used for formulating expressions in Mule.
+Within fields of many message processing components, in addition to the
+Transform Message component, you can use DataWeave to access, manipulate, 
+transform, and extract data from the Mule event.
+
+At runtime, Mule evaluates expressions while executing a flow to:
+
+* Extract data needed to process the current message.
+* Set or manipulate a value in the message.
+
+For more on this topic, see xref:dataweave.adoc[DataWeave Language].
+
+== Batch Processing
+
+Mule supports batch jobs, which split messages into individual records, 
+act on each record, then report the results and, if necessary, push the processed 
+output to other systems or queues. This functionality is particularly useful when 
+working with streaming input or when engineering "near real-time" data integration 
+between SaaS applications.
+
+Studio and Design Center provide various batch components that you can use to 
+configure batch processes. For more on this topic, see 
+xref:batch-processing-concept.adoc[Batch Processors].
+
+== Non-blocking Execution Engine
+
+The Mule execution engine is based on a non-blocking runtime. This is a
+task-oriented execution model that allows you to take advantage of non-blocking
+IO calls and to avoid performance problems.
+
+Flows always function synchronously. If you need to achieve asynchronous
+patterns, such as Fire-and-Forget, you can use the Async (`<async>`) component.
+
+Mule Event processors can indicate to the runtime whether they are CPU intensive,
+CPU light, or IO intensive operations. These workload types help the runtime tune 
+for different workloads, removing the need for you to manage thread pools manually 
+to achieve optimum performance. Instead, the runtime introspects the available
+resources (such as memory and CPU cores) in the system to tune thread pools 
+automatically. 
+
+=== Thread Pool Executor (Schedulers) Configuration
+
+You can configure schedulers (thread pool executors) through the
+well-commented `schedulers.conf` file in the standalone Mule EE
+installation.
```

Comment:
I think since we are not linking to any docs here (we SHOULD have documentation on this) then we should explain a bit more, like saying you can modify the formulas for the different kind of pools sizes for example.